packages/node_modules/@webex/plugin-meetings/test/integration/spec/journey.js

```diff
@@ -286,6 +297,80 @@ skipInNode(describe)('plugin-meetings', () => {
           throw e;
         }));
 
+      // FHD - remote video tests - starts
+      // Firefox bug - https://bugzilla.mozilla.org/show_bug.cgi?id=1537986
+      if (getBrowserName() !== 'Firefox') {
+        const getRemoteHeight = (minHeight = 360, maxHeight = 1080) => new Promise((resolve, reject) => {
+          testInterval = setInterval(() => {
+            const remoteSettings = bob.meeting.mediaProperties.remoteVideoTrack.getSettings();
+            const height = remoteSettings.height && +remoteSettings.height;
+
+            console.log(`Resolution for ${minHeight} ${maxHeight} : `, count, height);
+            if (height >= minHeight && height <= maxHeight) {
+              clearInterval(testInterval);
+              resolve(height);
+            }
+
+            if (count === INTERVAL_COUNT) {
+              clearInterval(testInterval);
+              if (!height) reject(new Error('Resolution undefined'));
+              else resolve(height);
+            }
+
+            count += 1;
+          }, WAIT_TIME);
+        });
+
+        it('alice resolution is 1080p, check what bob receives', async () => {
+          await testUtils.delayedPromise(alice.meeting.setLocalVideoQuality('1080p'));
+          const height = await getRemoteHeight(1080);
+
+          assert.isAbove(height, 500);
+          assert.isBelow(height, 1081);
+        });
+
+        it('bob requests 480p, should get resolution around 480p', async () => {
+          await testUtils.delayedPromise(bob.meeting.setRemoteQualityLevel('LOW'));
+          const height = await getRemoteHeight(400, 480);
+
+          assert.isAbove(height, 400);
+          assert.isBelow(height, 481);
+        });
+
+        it('bob upgrades remote to 720p, should get resolution around 720', async () => {
+          await testUtils.delayedPromise(bob.meeting.setRemoteQualityLevel('MEDIUM'));
+          const height = await getRemoteHeight(720, 1079);
+
+          assert.isAbove(height, 500);
+          assert.isBelow(height, 721);
+        });
+
+        it('bob requests 1080p, should get 1080 or max supported resolution', async () => {
+          await testUtils.delayedPromise(bob.meeting.setRemoteQualityLevel('HIGH'));
+          const height = await getRemoteHeight(721);
+
+          assert.isAbove(height, 719);
+          assert.isBelow(height, 1081);
+        });
+
+        it('alice resolution is 720p, check what bob receives', async () => {
+          await testUtils.delayedPromise(alice.meeting.setLocalVideoQuality('720p'));
+          const height = await getRemoteHeight(600, 720);
```

Comment:
why the mismatch between the values here and in the asserts later?

Reply:
The range provided here are preferred one, we may or may not get the value exactly in that range (Also mentioned [here](https://github.com/webex/webex-js-sdk/pull/2446#discussion_r985610199)). It's to try to get values as close as possible but `getRemoteHeight()` is not guaranteed to return value specifically in that range all the time. console.log is added inside the getRemoteHeight(), you may increase the timeout in the config file and check the  log for longer time and observe value changes gradually and sometimes takes longer than expected.

Reply:
At the end of the day the test can either pass or fail, there is nothing else in-between. No one will look at any logs for a passing test, so if the test can pass with resolution not actually being changed to the more stringent checks inside the getRemoteHeight() I see no point in doing the different boundary values between the call to getRemoteHeight() and asserts after it. With the code as it is, if it fails to change the resolution, the test will still pass but it will take longer, because getRemoteHeight() will wait for the higher res, but assert will pass because it has less stringent condition - so that's pointless, we're just wasting time waiting for getRemoteHeight() to finish.